forsys/general_matrix.py

```diff
@@ -42,111 +40,93 @@ def solve_system(self, **kwargs) -> list:
         assert self.lhs_matrix is not None, "LHS matrix not set"
         assert self.rhs_matrix is not None, "RHS matrix not set"
 
-        self.lhs_matrix, self.rhs_matrix = self.get_least_squares_matrices(self.lhs_matrix, 
-                                                                           self.rhs_matrix)
+        lhs_matrix_ls = self.lhs_matrix.T @ self.lhs_matrix
+        rhs_matrix_ls = self.lhs_matrix.T @ self.rhs_matrix  # vector

```

Comment:
I took the liberty of not replacing the original matrices stored as instance attributes with the least-squares transformed ones. Usually, storing different things under the same property in this way can lead to bugs.